cluster-autoscaler/cloudprovider/clusterapi/clusterapi_nodegroup.go

```diff
@@ -62,79 +59,28 @@ func (ng *nodegroup) MaxSize() int {
 }
 
 func (ng *nodegroup) Replicas() int {
-	if ng.MachineSet.Spec.Replicas == nil {
-		return 0
-	}
-	glog.Infof("machineset: %q has %d replicas", ng.MachineSet.Name, *ng.MachineSet.Spec.Replicas)
-	return int(*ng.MachineSet.Spec.Replicas)
+	return int(ng.replicas)
 }
 
 func (ng *nodegroup) SetSize(nreplicas int) error {
-	ms, err := ng.clusterapi.MachineSets(ng.MachineSet.Namespace).Get(ng.MachineSet.Name, v1.GetOptions{})
+	machineSet, err := ng.provider.clusterapi.MachineSets(ng.namespace).Get(ng.name, v1.GetOptions{})
```

Comment:
You should not access method of indirect field. It makes components more coupled. `ng.provider.clusterapiClient().MachineSets()` is better.

Reply:
This is all the same package. There's intrinsic coupling.

Reply:
Done (via dependency injection).